js/card-popup.js

```diff
@@ -0,0 +1,80 @@
+import {getSimilarArray, SIMILAR_APARTMENTS_COUNT} from './data.js';
+
+const cardPopupTemplate = document.querySelector('#card').content.querySelector('.popup');
+const map = document.querySelector('.map__canvas');
+cardPopupTemplate.classList.remove('hidden');
+const TYPE_NAME = {
+  'flat': 'Квартира',
+  'bungalow': 'Бунгало',
+  'house': 'Дом',
+  'palace': 'Дворец',
+  'hotel': 'Отель',
+};
+const dataCard = getSimilarArray(SIMILAR_APARTMENTS_COUNT);
+const dataCardsElements = [];
+dataCard.forEach(({author, offer}) => {
+  const dataElement = cardPopupTemplate.cloneNode(true);
+  const popupTitle = dataElement.querySelector('.popup__title');
+  const popupTextAddress = dataElement.querySelector('.popup__text--address');
+  const popupTextPrice = dataElement.querySelector('.popup__text--price');
+  const popupType = dataElement.querySelector('.popup__type');
+  const popupTextCapacity = dataElement.querySelector('.popup__text--capacity');
+  const popupTextTime = dataElement.querySelector('.popup__text--time');
+  const popupFeatures = dataElement.querySelector('.popup__features');
+  const popupDescription = dataElement.querySelector('.popup__description');
+  const popupAvatar = dataElement.querySelector('.popup__avatar');
+  const popupPhotos = dataElement.querySelector('.popup__photos');
+
+  popupTitle.textContent = offer.title;
+  popupTextAddress.textContent = offer.address;
+  popupTextPrice.textContent = `${offer.price} ₽/ночь`;
+  popupType.textContent = TYPE_NAME[offer.type];
+  popupTextCapacity.textContent = `${offer.rooms} комнаты для ${offer.guests} гостей`;
+  popupTextTime.textContent = `Заезд после ${offer.checkin}, выезд до ${offer.checkout}`;
+  popupFeatures.textContent = offer.features;
+  popupDescription.textContent = offer.description;
+  offer.photos.forEach((photo) => {
```

Comment:
offer тут по идее тоже можно распаковать

Как-то так

```suggestion
  const {
    address,
    checkin,
    checkout,
    description,
    features,
    guests,
    photos,
    price,
    rooms,
    title,
    type
  } = offer

  popupTitle.textContent = title;
  popupTextAddress.textContent = address;
  popupTextPrice.textContent = `${price} ₽/ночь`;
  popupType.textContent = TYPE_NAME[type];
  popupTextCapacity.textContent = `${rooms} комнаты для ${guests} гостей`;
  popupTextTime.textContent = `Заезд после ${checkin}, выезд до ${checkout}`;
  popupFeatures.textContent = features;
  popupDescription.textContent = description;
  photos.forEach((photo) => {
```